Source/Other/Extensions/DateExtension.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  DateExtention.swift
+//  Sequel Ace
+//
+//  Created by James on 3/11/2020.
+//  Copyright © 2020 Sequel-Ace. All rights reserved.
+//
+
+import Foundation
+
+extension Date {
+	
+	public func format(_ format: String, locale: Locale, timeZone: TimeZone) -> String {
```

Comment:
`formatWithFormat` (in Obj-C) looks weird to me... and doesn't indicate it returns String... what about `public func string(format: String, locale: Locale = .current, timeZone: TimeZone = .current) -> String`?

Not also the `locale: Locale = .current, timeZone: TimeZone = .current` - it means you don't need the functions above as there is default value so in swift one function servers one / two / three params. In Obj-C, you will need to fill all three, but for the future this is better to fit into one function :)

Reply:
updated